File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.java
class Solution {
    public int strStr(String haystack, String needle) {
        int n = haystack.length() , m = needle.length();

        for(int i = 0 ; i <= n - m ; i++){
            if( haystack.substring(i , i + m).equals(needle) ){
                return i;
            }
            // System.out.println(haystack.substring(i , i + m));
        }

        return -1;
    }
}

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.go
func strStr(haystack string, needle string) int {
    h := len(haystack)
    n := len(needle)
    switch {
        case n == 0:
            return 0
        case n == 1:
            for i := 0 ; i < h ; i++ {
                if string(haystack[i]) == needle {
                    return i
                }
            }
        case n == h:
            if haystack == needle {
                return 0
            }
        case n > h:
            return -1
        default:
            for i := 0 ; i < h - n + 1 ; i++ {
                if haystack[i : i + n] == needle {
                    return i
                }
            }
    }
    return -1
}